lerobot/common/robot_devices/robots/koch.py

```diff
@@ -0,0 +1,423 @@
+    print("      DRIVE_MODE: ", " ".join([str(i) for i in drive_mode]))
+    print("=====================================")
+
+    return homing_offset, drive_mode
+
+
+########################################################################
+# Alexander Koch robot arm
+########################################################################
+
+
+@dataclass
+class KochRobotConfig:
+    """
+    Example of usage:
+    ```python
+    KochRobotConfig()
+    ```
+    """
+
+    # Define all the components of the robot
+    leader_arms: dict[str, MotorsBus] = field(
+        default_factory=lambda: {
+            "main": DynamixelMotorsBus(
+                port="/dev/tty.usbmodem575E0031751",
+                motors={
+                    # name: (index, model)
+                    "shoulder_pan": (1, "xl330-m077"),
+                    "shoulder_lift": (2, "xl330-m077"),
+                    "elbow_flex": (3, "xl330-m077"),
+                    "wrist_flex": (4, "xl330-m077"),
+                    "wrist_roll": (5, "xl330-m077"),
+                    "gripper": (6, "xl330-m077"),
+                },
+            ),
+        }
+    )
+    follower_arms: dict[str, MotorsBus] = field(
+        default_factory=lambda: {
+            "main": DynamixelMotorsBus(
+                port="/dev/tty.usbmodem575E0032081",
+                motors={
+                    # name: (index, model)
+                    "shoulder_pan": (1, "xl430-w250"),
+                    "shoulder_lift": (2, "xl430-w250"),
+                    "elbow_flex": (3, "xl330-m288"),
+                    "wrist_flex": (4, "xl330-m288"),
+                    "wrist_roll": (5, "xl330-m288"),
+                    "gripper": (6, "xl330-m288"),
+                },
+            ),
+        }
+    )
+    cameras: dict[str, Camera] = field(default_factory=lambda: {})
+
+
+class KochRobot:
+    """Tau Robotics: https://tau-robotics.com
+
+    Example of usage:
+    ```python
+    robot = KochRobot()
+    robot.connect()
+    while True:
+        robot.teleop_step()
+    ```
+    """
+
+    def __init__(
+        self,
+        config: KochRobotConfig | None = None,
+        calibration_path: Path = ".cache/calibration/koch.pkl",
+        **kwargs,
+    ):
+        if config is None:
+            config = KochRobotConfig()
+        # Overwrite config arguments using kwargs
+        self.config = replace(config, **kwargs)
+        self.calibration_path = Path(calibration_path)
+
+        self.leader_arms = self.config.leader_arms
+        self.follower_arms = self.config.follower_arms
+        self.cameras = self.config.cameras
+        self.is_connected = False
+        self.logs = {}
+
+    def connect(self):
+        if self.is_connected:
+            raise ValueError(f"KochRobot is already connected.")
+
+        for name in self.follower_arms:
+            self.follower_arms[name].connect()
+            self.leader_arms[name].connect()
+
```

Comment:
place `self.is_connected=True` here ? otherwise running the calibration throws an error : 
```
if not self.is_connected:
            raise ValueError(f"KochRobot is not connected. You need to run `robot.connect()`.")
```

Reply:
Done

```python
def connect(self):
        if self.is_connected:
            raise RobotDeviceAlreadyConnectedError(
                "KochRobot is already connected. Do not run `robot.connect()` twice."
            )